libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -389,8 +389,12 @@ void PbftManager::resetPbftConsensus(PbftRound round) {
   LOG(log_dg_) << "Reset PBFT consensus to: period " << getPbftPeriod() << ", round " << round << ", step 1";
 
   // Reset broadcast counters
-  broadcast_votes_counter_ = 1;
-  rebroadcast_votes_counter_ = 1;
+  broadcast_soft_next_votes_counter_ = 1;
+  rebroadcast_soft_next_votes_counter_ = 1;
+  if (round == 1) {
```

Comment:
all of the code inside these multiple round == 1 could go into advancePeriod() function without any if